File: webui/src/main/scala/com/taskadapter/reporting/FieldMappingFormatter.scala
package com.taskadapter.reporting

import com.google.common.base.Strings
import com.taskadapter.connector.definition.FieldMapping
import com.taskadapter.model.Field

object FieldMappingFormatter {
  def format(mappings: Seq[FieldMapping[_]]): String = {
    mappings.map(m => {
      val field1 = formatField(m.fieldInConnector1)
      val field2 = formatField(m.fieldInConnector2)
      s"$field1 - $field2 selected: ${m.selected} default: ${m.defaultValue}"
    })
      .mkString(System.lineSeparator())
  }

  def formatField(field: Option[Field[_]]): String = {
    val string = field.map(f => {
      if (f.getClass.getSimpleName.startsWith("Custom")) {
        f.toString
      } else {
        f.getClass.getSimpleName.replace("$", "")
      }
    }).getOrElse("None")

    Strings.padEnd(string, 30, ' ')
  }
}

File: webui/src/main/java/com/taskadapter/reporting/FieldMappingFormatter.java
package com.taskadapter.reporting;

import com.google.common.base.Strings;
import com.taskadapter.common.ui.FieldMapping;
import com.taskadapter.connector.common.FieldPrettyNameBuilder;
import com.taskadapter.model.Field;

import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

public class FieldMappingFormatter {
    public static String format(List<FieldMapping<?>> mappings) {
        return mappings.stream().map(m -> {
            var field1 = formatField(m.getFieldInConnector1());
            var field2 = formatField(m.getFieldInConnector2());
            return String.format("%1$s - %2$s selected: %3$s default: %4$s",
                    field1, field2, m.isSelected(), m.getDefaultValue());
        }).collect(Collectors.joining(System.lineSeparator()));
    }

    private static String formatField(Optional<? extends Field<?>> field) {
        var string = field.map(FieldPrettyNameBuilder::getPrettyFieldName).orElse("None");
        return Strings.padEnd(string, 30, ' ');
    }
}
